thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/function/AbstractModularizedAnomalyFunction.java

```diff
@@ -270,4 +284,39 @@ public AnomalyTimelinesView getTimeSeriesView(MetricTimeSeries timeSeries,
     return this.getTimeSeriesView(anomalyDetectionContext, bucketMillis, metric, viewWindowStartTime, viewWindowEndTime,
         knownAnomalies);
   }
+
+  @Override
```

Comment:
whats the contract on these methods. Please add java docs for these methods. Are these methods called only once or multiple times. Can the implementations override these methods. Can they return a new instance every time getXModel is invoked?.

If this is one time, then I suggest renaming them to init or create or setup - something that indicates that this will be called only once when the Function is being setup.

Reply:
Thanks for the comments. I'll add java docs to these methods. This methods return a user-defined models. These methods are usually called only once. They return the same instance every time they are called. The implementation can override these methods, but in realistic, they are seldom override by the implementations. That's why I want to move it forward to the abstract class. As the methods only return the pre-setup models (the models are created and initialized in init()), using init may disguise these methods.